elements/pfe-spinner/BaseSpinner.ts

```diff
@@ -0,0 +1,60 @@
+import { LitElement, html } from 'lit';
+import { property } from 'lit/decorators.js';
+import { observed } from '@patternfly/pfe-core/decorators.js';
+
+import styles from './BaseSpinner.scss';
+
+export type SpinnerSize = (
+  | 'sm'
+  | 'md'
+  | 'lg'
+  | 'xl'
+)
+
+/**
+ * Base spinner class
+ *
+ * @cssprop {<length>} --pf-c-spinner--diameter                      {@default `3.375rem`}
+ * @cssprop {<length>} --pf-c-spinner--Width                         {@default `3.375rem`}
+ * @cssprop {<length>} --pf-c-spinner--Height                        {@default `3.375rem`}
+ * @cssprop {<color>}  --pf-c-spinner--Color                         {@default `#06c`}
+ * @cssprop {<length>} --pf-c-spinner--m-sm--diameter                {@default `0.625rem`}
+ * @cssprop {<length>} --pf-c-spinner--m-md--diameter                {@default `1.125rem`}
+ * @cssprop {<length>} --pf-c-spinner--m-lg--diameter                {@default `1.5rem`}
+ * @cssprop {<length>} --pf-c-spinner--m-xl--diameter                {@default `3.375rem`}
+ * @cssprop {<time>}   --pf-c-spinner--AnimationDuration             {@default `1.4s`}
+ * @cssprop {<string>} --pf-c-spinner--AnimationTimingFunction       {@default `linear`}
+ * @cssprop {<number>} --pf-c-spinner--stroke-width                  {@default `10`}
+ * @cssprop {<color>}  --pf-c-spinner__path--Stroke                  {@default `#06c`}
+ * @cssprop {<number>} --pf-c-spinner__path--StrokeWidth             {@default `10`}
+ * @cssprop {<string>} --pf-c-spinner__path--AnimationTimingFunction {@default `ease-in-out`}
+ */
+
+export abstract class BaseSpinner extends LitElement {
+  static readonly styles = [styles];
+
+  @property({ reflect: true }) role = 'progressbar';
+
+  /** Size variant of progress */
+  @property({ reflect: true }) size: SpinnerSize = 'xl';
+
+  /** Custom diameter of spinner set as CSS variable */
+  @observed
+  @property({ reflect: true }) diameter?: string;
+
+  override render() {
+    return html`
+      <svg class="pf-c-spinner" viewBox="0 0 100 100">
+        <circle class="pf-c-spinner__path" cx="50" cy="50" r="45" fill="none" />
+      </svg>
+    `;
+  }
+
+  protected _diameterChanged() {
+    if (!this.diameter) {
+      return;
+    }
+
+    this.style.setProperty('--pf-c-spinner--diameter', this.diameter);
```

Comment:
this could potentially be set internally with a `styleMap`. If it's the case that we always want the diameter attribute to override the css cascade, then we should in fact do it that way

```js
html`<svg style="${styleMap({ '--pf-c-spinner--diameter': this.diameter })">`
```

Reply:
I've probably spent more time than it's worth trying to get the wrapper `pfe-spinner` to work with your suggestion. I kept running into issues where the wrapper component wouldn't shrink to the size of the svg which could lead to some spacing issues. I think we can leave it as it is for right now and make an adjustment later if we need to.